src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java

```diff
@@ -78,7 +73,10 @@ public IndexShardGateway(ShardId shardId, @IndexSettings Settings indexSettings,
 
         this.waitForMappingUpdatePostRecovery = indexSettings.getAsTime("index.gateway.wait_for_mapping_update_post_recovery", TimeValue.timeValueSeconds(30));
         syncInterval = indexSettings.getAsTime("index.gateway.sync", TimeValue.timeValueSeconds(5));
-        if (syncInterval.millis() > 0) {
+        if (indexShard instanceof ShadowIndexShard) {
```

Comment:
Since you have access to the `IndexShard` you can use `indexShard.routingEntry().primary() == false && IndexMetaData.isIndexUsingShadowReplicas(indexSettings)` for this.